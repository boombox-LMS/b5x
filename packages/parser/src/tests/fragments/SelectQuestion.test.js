const {
  exampleSingleSelectQuestionMarkup,
  exampleMultiSelectQuestionMarkup,
  SelectQuestionDataSchema,
} = require("../../../dist/componentBuilders/fragments/parsingClasses/SelectQuestion");
const { RawFragmentSchema } = require("@b5x/types");

describe("The SelectQuestion fragment returns the expected data for single-select questions", () => {
  const fragment = global.buildFragmentFromMarkupString(
    exampleSingleSelectQuestionMarkup
  );
  const rawFragment = fragment.packageForApi();

  test("The parsed single-select-question fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed single-select-question fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });

  test("The parsed single-select-question fragment uses the expected data schema", () => {
    const validator = () => {
      SelectQuestionDataSchema.parse(rawFragment.data);
    };
    expect(validator).not.toThrowError();
  });
});

describe("The SelectQuestion fragment returns the expected data for multi-select questions", () => {
  const fragment = global.buildFragmentFromMarkupString(
    exampleMultiSelectQuestionMarkup
  );
  const rawFragment = fragment.packageForApi();

  test("The parsed multi-select-question fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed multi-select-question fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });

  test("The parsed single-select-question fragment uses the expected data schema", () => {
    const validator = () => {
      SelectQuestionDataSchema.parse(rawFragment.data);
    };
    expect(validator).not.toThrowError();
  });
});
