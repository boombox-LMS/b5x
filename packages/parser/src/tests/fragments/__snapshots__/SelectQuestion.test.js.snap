// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SelectQuestion fragment returns the expected data for multi-select questions The parsed multi-select-question fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
  ],
  "contentType": "MultiSelectQuestion",
  "contents": "",
  "data": Object {
    "choices": Array [
      Object {
        "contents": "Pink",
        "value": "pink",
      },
      Object {
        "contents": "Green",
        "value": "green",
      },
      Object {
        "contents": "Blue",
        "value": "blue",
      },
      Object {
        "contents": "Purple",
        "value": "purple",
      },
      Object {
        "contents": "Yellow",
        "value": "yellow",
      },
      Object {
        "contents": "Orange",
        "value": "orange",
      },
    ],
  },
  "dependencyUris": Array [],
  "digest": "2d7599c6e1289966b462e2f79e50b4d1",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": true,
  "isStateful": true,
  "uri": "test-fragment-uri",
}
`;

exports[`The SelectQuestion fragment returns the expected data for single-select questions The parsed single-select-question fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
  ],
  "contentType": "SingleSelectQuestion",
  "contents": "",
  "data": Object {
    "choices": Array [
      Object {
        "contents": "the Midwest",
        "value": "midwest",
      },
      Object {
        "contents": "the South",
        "value": "south",
      },
      Object {
        "contents": "the East Coast",
        "value": "east-coast",
      },
      Object {
        "contents": "the West Coast",
        "value": "west-coast",
      },
    ],
  },
  "dependencyUris": Array [],
  "digest": "36689c382439e6bf42c91de3316cb903",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": true,
  "isStateful": true,
  "uri": "test-fragment-uri",
}
`;
